### DISCUSSION TITLE
How to use "cordova-plugin-idfa" in Vue.js (Quasar) application?

### DISCUSSION THREAD

**Discussion by tesicg** [quasarframework/quasar#9690] [General - Components / Directives / etc] [answered]
We have Vue.js (Quasar) mobile app that is published on App Store. Start from iOS 14.5+, it's mandatory when first time start the app to ask user for permission to access its personal data. There is standard dialog, which is part of iOS that should be invoked for that purpose.

We've found "Cordova plugin for getting Advertising ID (IDFA or AAID)" - cordova-plugin-idfa that should help us. Here is the link:

https://github.com/chemerisuk/cordova-plugin-idfa

There is some code example, but we don't understand where to put that code in our application and how to use it.

We need help about this issue. Thank you in advance.

**Comment by ontwikkelfabriek** [comment, marked as answer]
@tesicg 

Presuming you know how to install cordova plugins....

Here are some examples of how to use some cordova plugins inside a Quasar project:
https://quasar.dev/quasar-cli/developing-cordova-apps/cordova-plugins#example-battery-status

I guess I would put the code in the mounted hook in the layout you use when you first start the app:

If you want to use this app also for Android or web, you'll have to use Quasar's platform detection api:
https://quasar.dev/options/platform-detection#properties

MainLayout.vue :
```javascript
<template>
...
</template>

<script>
const idfaPlugin = cordova.plugins.idfa;
...
export default {
   ...
   mounted () {
     idfaPlugin.getInfo()
        .then(info => {
            if (!info.trackingLimited) {
                return info.idfa || info.aaid;
            } else if (info.trackingPermission === idfaPlugin.TRACKING_PERMISSION_NOT_DETERMINED) {
                return idfaPlugin.requestPermission().then(result => {
                    if (result === idfaPlugin.TRACKING_PERMISSION_AUTHORIZED) {
                        return idfaPlugin.getInfo().then(info => {
                            return info.idfa || info.aaid;
                        });
                    }
                });
            }
        })
        .then(idfaOrAaid => {
            if (idfaOrAaid) {
                console.log(idfaOrAaid);
            }
        });
   }
}
...
</script>

```

**Reply by tesicg** [reply]
I did what you suggested in our MainLayout.vue component, but there's an error when run the app. Please, take a look at the screenshots before and after adding plugin code:

Before (the app works fine):

https://drive.google.com/file/d/1WfdOlhZ8qnS-mlagOHuB87R76Ynf0RSU/view?usp=sharing

After (the app doesn't work fine, there's an white screen on mobile device):

https://drive.google.com/file/d/1KdGyVxPbXtCXQwd3X4_VjHxNPZV6qt9a/view?usp=sharing

**Reply by ontwikkelfabriek** [reply]
@tesicg 

Well it can't find the `cordova` object. It's probably handing  out in `window.cordova`.

Try this:
```
const idfaPlugin = window.cordova.plugins.idfa;
```

If it's not than you'll have to debug to search for the `cordova` object .... it's handing out somewhere.
